internal/controller/numaresourcesoperator_controller.go

```diff
@@ -524,6 +524,19 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 		}
 	}
 
+	r.RTEManifests.Core.DaemonSet.Spec.Template.Spec.Affinity = &corev1.Affinity{
+		PodAntiAffinity: &corev1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: r.RTEManifests.Core.DaemonSet.Spec.Template.Labels,
+					},
+					TopologyKey: "kubernetes.io/hostname",
+				},
+			},
+		},
+	}
+
```

Comment:
this looks correct but should be a helper function in `pkg/objectupdate/rte`. Considered you added a controller test, unit tests are welcome but not needed.

If there is a suitable existing helper in `pkg/objectupdate/rte` feel free to add this code there.